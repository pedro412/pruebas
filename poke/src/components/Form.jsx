import { useState } from 'react';

const apiURL = 'http://localhost:4000/api/v1/pokemon';

async function fetchPokemon(name) {
  try {
    const result = await fetch(`${apiURL}/${name}`);
    const data = await result.json();
    return data.pokemon;
  } catch (error) {
    return null;
  }
}

const Form = ({ setPokemons }) => {
  const [value, setValue] = useState('');

  const handleSubmit = async (e) => {
    e.preventDefault();
    const pokemon = await fetchPokemon(value);
    if (pokemon) {
      pokemon.count = 1;
      setPokemons(pokemon);
    }
  };

  return (
    <>
      <h1>Form</h1>
      <form onSubmit={handleSubmit} id="form">
        <input
          name="name"
          onChange={(e) => setValue(e.target.value)}
          value={value}
          type="text"
        />
        <button>Catch</button>
      </form>
    </>
  );
};

export default Form;
