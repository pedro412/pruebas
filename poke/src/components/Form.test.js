import Form from './Form';
const enzyme = require('enzyme');

describe('<Form />', () => {
  const wrapper = enzyme.shallow(<Form />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change input', () => {
    const wrapper = enzyme.mount(<Form />);

    wrapper.find('input').simulate('change', {
      target: {
        name: 'name',
        value: 'pikachu',
      },
    });

    expect(wrapper.find('input').prop('value')).toBe('pikachu')
  });
});
