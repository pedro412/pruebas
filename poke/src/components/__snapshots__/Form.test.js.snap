// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render correctly 1`] = `
<Fragment>
  <h1>
    Form
  </h1>
  <form
    id="form"
    onSubmit={[Function]}
  >
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button>
      Catch
    </button>
  </form>
</Fragment>
`;
